### DISCUSSION TITLE
How to parse top-level JSON arrays/lists with cel-go?

### DISCUSSION THREAD

**Discussion by mrueg** [google/cel-go#864] [Q&A]
I've asked [here](https://github.com/undistro/cel-playground/issues/44) and thought it might be good to ask here as well.

Does anyone have a minimal example how to parse and execute CEL against top-level JSON arrays?

```json
[
  {
    "noun": "lion",
    "population": 123,
    "predator": true
  },
  {
    "noun": "deer",
    "population": 456,
    "predator": false
  },
  {
    "noun": "pigeon",
    "population": 789,
    "predator": false
  }
]
```

with jsonpath I could query a key like the following: `{ [*].predator }`

What is the CEL equivalent to do that?

Here's the eval function that is used in cel-playground where I came across the problem: https://github.com/undistro/cel-playground/blob/main/eval/eval.go#L51

**Comment by mrueg** [comment]
@TristonianJones apologies for bugging you here directly, are you using discussions in this project or is it better to move this into an issue?

**Comment by TristonianJones** [comment]
@mrueg not a problem, thanks for moving this to a discussion. Simply put, CEL is not jsonpath. The conveniences of jsonpath are well-suited to JSON, but not necessarily something we want to support in CEL; however, you could expose a jsonpath function which would allow you to blend the two together.

The expression you provided, the JSON literal, is a valid CEL literal. To express something equivalent in CEL, you'd probably do the following: `values.map(e, e.predator)` which would return a list of all of the predator bool values. If the jsonpath is only expected to return records where `predator == true`, then this would be the expression you'd want: `values.filter(e, e.predator)`

**Reply by mrueg** [reply]
Thanks for your response @TristonianJones ! I understand. 
If I did not want to apply any function and the top level array is a valid CEL literal, is there an idempotent operation in CEL that I could use? 

Context: The result of the query will be used in another query (so in JSONPath it would be [*]).
